### PR TITLE
[EA Forum only] add field for users to allow us to use their votes for content recommendations

### PR DESCRIPTION
**Pull request by s-cheng**
The EA Forum wants to allow users to opt in to letting us use their votes in aggregate with other users' votes to inform our content recommendation choices. For example, if we get >50 users opting in to this, we might show their aggregate votes on the [digest planner table](https://github.com/ForumMagnum/ForumMagnum/pull/7403) to help us make decisions on which posts to include in the digest.

This PR just adds the field, since we haven't totally finalized how we will use it yet.

Screen Shot 2023-07-12 at 5 37 14 PM

### CODE REVIEW

File: packages/lesswrong/lib/collections/users/schema.ts [hunk 1]
```diff
@@ -2630,6 +2630,19 @@ const schema: SchemaType<DbUser> = {
...
+  allowAdminsToUseVotes: {
+    type: Boolean,
+    optional: true,
+    nullable: true,
+    hidden: forumTypeSetting.get() !== 'EAForum',
+    canRead: [userOwns, 'admins'],
+    canUpdate: [userOwns, 'sunshineRegiment', 'admins'],
+    canCreate: ['members'],
+    label: "Allow the EA Forum Team to use my votes for content recommendations",
+    tooltip: "Allow us to use your votes in aggregate with other users' votes to inform our content recommendation choices, such as in the weekly digest. We will never look at your votes on an individual level.",
```
Comment (b0b3rt) [inline packages/lesswrong/lib/collections/users/schema.ts:2642]: > We will never look at your votes on an individual level

If this is false in e.g. the case of investigating abusive voting behavior, then probably wants a slight rewording.

Comment (oetherington) [reply packages/lesswrong/lib/collections/users/schema.ts:2642]: Also, we already do this for recommendations on the post page. We either need to reword this to not include that, or we need to make recommendations on the post page aware of this option.

**Review by oetherington — APPROVED** [review]
Looks good, subject to comments
